eks/deploy.go

```diff
@@ -52,39 +58,54 @@ func RollOutDeployment(
 	logger.Infof("Successfully authenticated with AWS")
 
 	// Retrieve the ASG object and gather required info we will need later
-	originalCapacity, currentInstanceIds, err := getAsgInfo(asgSvc, eksAsgName)
+	asgInfo, err := getAsgInfo(asgSvc, eksAsgName)
 	if err != nil {
 		return err
 	}
 
 	// Calculate default max retries
 	if maxRetries == 0 {
-		maxRetries = getDefaultMaxRetries(originalCapacity, sleepBetweenRetries)
+		maxRetries = getDefaultMaxRetries(asgInfo.originalCapacity, sleepBetweenRetries)
 		logger.Infof(
 			"No max retries set. Defaulted to %d based on sleep between retries duration of %s and scale up count %d.",
 			maxRetries,
 			sleepBetweenRetries,
-			originalCapacity,
+			asgInfo.originalCapacity,
 		)
 	}
 
 	// Make sure ASG is in steady state
-	if originalCapacity != int64(len(currentInstanceIds)) {
+	if asgInfo.originalCapacity != int64(len(asgInfo.currentInstanceIDs)) {
 		logger.Infof("Ensuring ASG is in steady state (current capacity = desired capacity)")
 		err = waitForCapacity(asgSvc, eksAsgName, maxRetries, sleepBetweenRetries)
 		if err != nil {
 			logger.Error("Error waiting for ASG to reach steady state. Try again after the ASG is in a steady state.")
 			return err
 		}
 		logger.Infof("Verified ASG is in steady state (current capacity = desired capacity)")
-		originalCapacity, currentInstanceIds, err = getAsgInfo(asgSvc, eksAsgName)
+		asgInfo, err = getAsgInfo(asgSvc, eksAsgName)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Make sure there is enough max size capacity to scale up
+	maxCapacityForUpdate := asgInfo.originalCapacity * 2
+	if asgInfo.maxSize < maxCapacityForUpdate {
+		defer func() {
+			err := setAsgMaxSize(asgSvc, eksAsgName, asgInfo.maxSize)
+			if err != nil && returnErr == nil {
+				returnErr = err
+			}
+		}()
```

Comment:
NIT: add a comment or extract into a function to make it clear that this is resetting the ASG back to its original max size

Reply:
Addressed in https://github.com/gruntwork-io/kubergrunt/pull/147